GSoC blog post 3: successful updates (#13)
ruma/ruma.github.io: content/news/2020-06-26-gsoc-update3-2020.md

+++
title = "Successful Updates"
date = 2020-06-26
extra.author = "Devin Ragotzy"
+++

> This week in the ruma/matrix Google Summer of Code project, ruma-events was made ready for use! After adding stripped and sync event generation to the [`event_enum!` macro](https://github.com/ruma/ruma/commit/fdf87a38a219933ffe99b17baf87841a79702f28) there were only a few small tweaks needed to try it out in some dependent rust crates. I spent a few days, converting [matrix-rust-sdk](https://github.com/matrix-org/matrix-rust-sdk/pull/63) to use the ruma-monorepo. Since ruma is used on both client and server-side I, also opened a PR to update [Conduit](https://git.koesters.xyz/timo/conduit/pulls/133) a homeserver implementation written in rust. To test that everything worked together, I updated [rumatui](https://github.com/DevinR528/rumatui/commit/1ae8776f9a59c297c8c9b2287f7c830ab67302b1), a command-line client written in rust. Then I could test that Conduit sent, and matrix-rust-sdk received the new ruma events successfully. During the updating, I felt the pain of not having accessor methods for the `Any*Event` enums to get at the event fields held within. I have opened a [pull request](https://github.com/ruma/ruma/pull/97) to add the generation of these methods to the `event_enums!` macro. Hopefully, these PR's will be merged and the ruma monorepo can be tested in the wild!

Reviewer: ```suggestion
This week in the ruma/matrix Google Summer of Code project, ruma-events was made ready for use! After adding stripped and sync event generation to the [`event_enum!` macro](https://github.com/ruma/ruma/commit/fdf87a38a219933ffe99b17baf87841a79702f28) there were only a few small tweaks needed to try it out in some dependent rust crates. I spent a few days converting [matrix-rust-sdk](https://github.com/matrix-org/matrix-rust-sdk/pull/63) to use the ruma-monorepo. Since ruma is used on both client and server-side I, also opened a PR to update [Conduit](https://git.koesters.xyz/timo/conduit/pulls/133), a homeserver implementation written in Rust. To test that everything worked together, I updated [rumatui](https://github.com/DevinR528/rumatui/commit/1ae8776f9a59c297c8c9b2287f7c830ab67302b1), my command-line client written in rust.

Then I could test that Conduit sent, and matrix-rust-sdk received the new ruma events successfully. While updating, I felt the pain of not having accessor methods for the `Any*Event` enums to get at the event fields held within. I have opened a [pull request](https://github.com/ruma/ruma/pull/97) to add the generation of these methods to the `event_enums!` macro. Hopefully, this PR will be merged and the ruma monorepo can be tested in the wild!
```

Suggestion re. rumatui is completely optional. Last one too, maybe I just missed something and you were trying to refer to more than the latest PR?

Reviewer: Diff highlighting broken apparently? 🤷

Author: Thanks I like the changes. I changed the last sentence to
> Hopefully, the Conduit and matrix-sdk PR's will be merged and the ruma monorepo can be tested in the wild!
